asapdiscovery-data/asapdiscovery/data/plip.py

```diff
@@ -0,0 +1,343 @@
+    if isinstance(sites, dict):
+        sites = [sites]
+
+    for bs in sites:
+        for _, data in bs["interactions"].items():
+            if data:
+                # we build keys for the dict to be unique, so no interactions are overwritten
+                for intn_type, intn_data in data.items():
+                    if isinstance(
+                        intn_data, list
+                    ):  # multiple interactions of this type
+                        for intn_data_i in intn_data:
+                            k, v = build_interaction_dict(
+                                intn_data_i,
+                                intn_counter,
+                                intn_type,
+                                color_method,
+                                protein,
+                                target,
+                            )
+                            intn_dict[k] = v
+                            intn_counter += 1
+
+                    elif isinstance(intn_data, dict):  # single interaction of this type
+                        k, v = build_interaction_dict(
+                            intn_data,
+                            intn_counter,
+                            intn_type,
+                            color_method,
+                            protein,
+                            target,
+                        )
+                        intn_dict[k] = v
+                        intn_counter += 1
+
+    return intn_dict
+
+
+## this should be placed around that area as well, but FINTscore should be added to docking scores by default
+def compute_fint_score(protein, pose, target):
+    """
+    Compute the Fitness Interaction Score (FINTscore) given a dict with interactions generated by PLIP.
+
+    Parameters
+    ----------
+    intn_dict : dict
+        Dictionary containing PLIP output with information on interactions that were found. Created by
+        asapdiscovery.dataviz.html_viz.HTMLVisualizer.get_interactions_plip().
+    fintscore_parameters : str
+        Path to a YAML file that contains settings for rewards/penalties of interaction types.
+
+    Returns
+    ----------
+    intn_score: float
+        Score based purely on interactions, without penalties applied
+    fint_score: float
+        FINTscore which is computed as the `intn_score` with penalties applied
+    """
+    # read YAML file that contains settings for rewards/penalties of interaction types.
+    fintscore_parameters = yaml.safe_load(Path(FINTSCORE_PARAMETERS).read_text())
+
+    # set empty parameters to add to when iterating over interactions.
+    penalty_multipliers = 1
+    reward_multipliers = 1
+    intn_score_bucket = []
+
+    # iterate over each interaction that was found.
+    intn_dict = get_interactions_plip(protein, pose, "fitness", target)
+
+    for _, data in intn_dict.items():
+        # if the interaction is with backbone, add a reward to the score.
+        if data["color"] == "#008000":
```

Comment:
Using the colour here is fragile as it is not strongly coupled with the code that generates the colours if you know what I mean. Is there another easy way to do this? If not happy to let slide for now as long as we make a note of it.

Reply:
Agreed, it also makes the `metadata` parameter yaml more opaque. I didn't see an obvious way of doing it more logically while I was building the MVP so I just took this route - will try to find a better one.

Reply:
Raised #785 to address this, but first lets get this MVP shipped and we can work on refining it.